### PR TITLE
Change signature of TranslationFinder::getTranslationFilesFromPath to accept null pattern

### LINKED ISSUE #23344: Error on translation when adding a new theme
This issue is related to the [Refactoring of Translation storage component](https://github.com/PrestaShop/PrestaShop/issues/19242) project

You can read about the project [here](https://github.com/PrestaShop/PrestaShop/issues/19242).

#### Describe the bug

After uploading a new theme and click on save there is an error 

#### Steps to Reproduce

1. Go to Bo -> Theme & logo
2. Click on add a new theme
3. Upload new theme and save

**Screenshots**

Capture d’écran 2021-02-22 à 12 18 15

#### Additional information

* PrestaShop version: develop
* PHP version: 7.1

### PR DESCRIPTION
**Pull request by sowbiba**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | TranslationFinder::getTranslationFilesFromPath arguments was untyped before https://github.com/PrestaShop/PrestaShop/commit/e80e74027275760c19950625c41292a0f2afb436#diff-4e7fbb6e634e3c056fc15d7dfc353f8b0d7f668166e289476c93e6b8fa2fcdcf that means that null values was allowed. When we typed them, we forgot to allow NULL for $pattern. This PR fixes that
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #23344 
| How to test?      | See the issue

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23857)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

File: src/PrestaShopBundle/Translation/Provider/TranslationFinder.php [hunk 1]
```diff
@@ -86,13 +86,13 @@ private function removeTrailingLocaleFromDomains(MessageCatalogueInterface $cata
...
      * @param string[] $paths
-     * @param string $pattern
+     * @param string|null $pattern
      *
      * @return Finder
      *
      * @throws FileNotFoundException
      */
-    private function getTranslationFilesFromPath(array $paths, string $pattern): Finder
+    private function getTranslationFilesFromPath(array $paths, ?string $pattern = null): Finder
```
Comment (jolelievre) [inline src/PrestaShopBundle/Translation/Provider/TranslationFinder.php:95]: Maybe we should add some unit test to avoid a regression?

Comment (eternoendless) [inline src/PrestaShopBundle/Translation/Provider/TranslationFinder.php:95]: This is a private method, so it's not a breaking change

Comment (sowbiba) [reply src/PrestaShopBundle/Translation/Provider/TranslationFinder.php:95]: Test added

**Review by jolelievre — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

File: tests/Integration/PrestaShopBundle/Translation/Provider/TranslationFinderTest.php [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            'ModulesCheckpaymentAdmin',
+            'ModulesCheckpaymentShop',
+            'ModulesWirepaymentAdmin',
+            'ModulesWirepaymentShop',
+            'ShopNotificationsWarning',
+            'messages',
+        ], $domains);
+    }
+
+    public function testGetCatalogueFromPathsFiltersDomainsIfPatternIsGiven()
```
Comment (Progi1984) [inline tests/Integration/PrestaShopBundle/Translation/Provider/TranslationFinderTest.php]:
```suggestion
    public function testGetCatalogueFromPathsFiltersDomainsIfPatternIsGiven(): void
```

File: tests/Integration/PrestaShopBundle/Translation/Provider/TranslationFinderTest.php [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    protected function setUp()
+    {
+        self::bootKernel();
+        $this->resourcesDir = self::$kernel->getContainer()->getParameter('test_translations_dir');
+
+        $this->finder = new TranslationFinder();
+    }
+
+    public function testGetCatalogueFromPathsReturnsAllDomainsIfNoPatternGiven()
```
Comment (Progi1984) [inline tests/Integration/PrestaShopBundle/Translation/Provider/TranslationFinderTest.php]:
```suggestion
    public function testGetCatalogueFromPathsReturnsAllDomainsIfNoPatternGiven(): void
```

File: tests/Integration/PrestaShopBundle/Translation/Provider/TranslationFinderTest.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    /**
+     * @var TranslationFinder
+     */
+    private $finder;
+    /**
+     * @var string
+     */
+    private $resourcesDir;
+
+    protected function setUp()
```
Comment (Progi1984) [inline tests/Integration/PrestaShopBundle/Translation/Provider/TranslationFinderTest.php]:
```suggestion
    protected function setUp(): void
```

File: tests/Integration/PrestaShopBundle/Translation/Provider/TranslationFinderTest.php [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace Tests\Integration\PrestaShopBundle\Translation\Provider;
```
Comment (Progi1984) [inline tests/Integration/PrestaShopBundle/Translation/Provider/TranslationFinderTest.php]:
```suggestion
 */

declare(strict_types=1);

namespace Tests\Integration\PrestaShopBundle\Translation\Provider;
```

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

**Review by Progi1984 — APPROVED** [review]

**Review by PululuK — APPROVED** [review]

**Comment by aDamelincourt** [comment]
Thanks @sowbiba 
✔️  QA approved
Themes were installed and uninstalled without any problem in BO

**Comment by Progi1984** [comment]
Thanks @sowbiba & @aDamelincourt

**Pull request merged**